.github/workflows/benchmark_trigger.yml

```diff
@@ -0,0 +1,113 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: Benchmark Trigger
+
+# Trigger benchmark CI workflows when benchmark PR labels are added.
+# When a benchmark label is added, this cancels and then re-runs the CI
+# workflow. When the workflow is re-run, it will pick up the latest PR
+# description and labels. See
+# https://github.com/openxla/iree/issues/10042#issuecomment-1449250094 for
+# more background.
+
+on:
+  pull_request_target:
+    types:
+      - labeled
+
+env:
+  PR_NUMBER: ${{ github.event.number }}
+  HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+  # Target workflow that runs the benchmarks.
+  WORKFLOW_NAME: CI
+  BENCHMARK_LABEL_PREFIX: 'benchmarks:'
+
+jobs:
+  # Precondition check is a separate step because we can't put the concurrency
+  # constraint on it; otherwise the irrelevant labeling events will cancel the
+  # events that meet the preconditions. Even with cancel-in-progress = false,
+  # the queued events will still be cancelled.
+  check:
+    runs-on: ubuntu-20.04
+    outputs:
+      should-rerun: ${{ steps.precondition.outputs.should-rerun }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: "Checking labels"
+        id: precondition
+        run: |
+          echo "should-rerun=$(jq --raw-output \
+            --arg label_prefix ${BENCHMARK_LABEL_PREFIX} \
+            '.label.name | startswith($label_prefix)' \
+            ${GITHUB_EVENT_PATH})" >> "${GITHUB_OUTPUT}"
+
+  trigger:
+    needs: check
+    if: needs.check.outputs.should-rerun == 'true'
+    runs-on: ubuntu-20.04
+    permissions:
+      actions: write
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    concurrency:
+      # This cancels queued and in-progress trigger runs for the same PR.
+      group: ${{ github.workflow }}-${{ github.event.number }}
```

Comment:
Are you sure this is right? I think you need `event.pull_request.number`, but honestly I screw this up every time. Maybe just print it out in the workflow and make sure it actaully returns anything

Reply:
Tested. `github.event.number` seems to be same as `event.pull_request.number` here.